javascript/extractor/src/com/semmle/js/parser/RegExpParser.java

```diff
@@ -499,7 +500,11 @@ private RegExpTerm parseCharacterClass() {
   private RegExpTerm parseCharacterClassElement() {
     SourceLocation loc = new SourceLocation(pos());
     RegExpTerm atom = this.parseCharacterClassAtom();
-    if (!this.lookahead("-]") && this.match("-"))
+    for (String c : Arrays.asList("d", "D", "s", "S", "w", "W")) {
```

Comment:
I think it would be prudent to be a little more efficient here. Creating a list and six string concatenations is quite a lot in order to parse a single character. Keep in mind that we speculatively parse strings as regexps, so this function is quite hot.

A simple first step is to start by checking for `-` and no more work if the next character is not `-`. Next we can store the string list `-\d`, `-\D`, `-\s`, `-\S`. `-\w`, `\-W` in a constant to simplify the full check.